### DISCUSSION TITLE
collapsable sections

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#3711] [Ideas]
Was it ever suggested to enable collapse-able section? When reading long pages it is useful to be able to collapse a certain section if that seems not important at the time of reading, instead of scrolling to the next one.

Below is the example of this function on one of docs sites I use:

![2022-03-14_12-06-24 (1)](https://user-images.githubusercontent.com/5679861/158160107-b9b9a86f-03ed-45a9-9bc4-52931fce1743.gif)

**Comment by ben519** [comment]
Nice post! (The visual is great.)

Note that this is kind of achievable with [collapsible admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/?h=admon#admonition-icons).

**Reply by hellt** [reply]
yes, collapsable admonitions are great for inlining information one wouldn't want to display by default.

This idea is more like a site-wide setting for headings

**Reply by ben519** [reply]
I understand the difference, but personally I don't think this feature "adds enough" to warrant development time. I'd rather see things like [blog features](https://github.com/squidfunk/mkdocs-material/issues/3353) and continued bug fixes get prioritized over this (sorry!). Hopefully others will chime in with their thoughts.

**Reply by hellt** [reply]
me too. This is just an idea that might be implemented some day and enrich the collection of features mkdocs-material has today

**Comment by squidfunk** [comment]
Thanks for suggesting. Could be achievable with some JS and CSS, but is out of scope for this project.

**Comment by lmmx** [comment]
Just in case people weren't aware, you can get this in mkdocs without changing anything by just wrapping the part you want to be collapsible inside a `details` tag after a `summary` tag

```
<details><summary>Click to show</summary>

The collapsed section's content goes here

</details>
```

![Screenshot from 2024-01-23 20-42-56](https://github.com/squidfunk/mkdocs-material/assets/2979452/2fb3e9ed-6757-4375-86c7-36a15bbaf013)

**Reply by squidfunk** [reply]
We're also looking into this as part of #6289.

**Reply by stejackson94** [reply]
I was hoping to have multiple of these on one page, but it seems as soon as i put one in, all the sections fall inside each other 

any ideas?

**Reply by TechxArtisan** [reply]
Where and how can I customize the color and icon of the summary tag? thx!

**Reply by TechxArtisan** [reply]
I'm answering to myself:

```
??? faq "This is a question"
      I'm saying something here. Below are break-down questions:
      <details class="faq">
         <summary>Click to show</summary>
         The collapsed section's content goes here
      </details>
      <details class="faq">
         <summary>Click to show</summary>
         The collapsed section's content goes here
      </details>
```

css:
```
:root {
    --md-admonition-icon--faq: url(/assets/faq.svg)}
  .md-typeset .admonition.faq,
  .md-typeset details.faq {
    border-color: #FF8D1A;
  }
  .md-typeset .faq > .admonition-title,
  .md-typeset .faq > summary {
    background-color: rgba(155, 120, 43, 0.1);
  }
  .md-typeset .faq > .admonition-title::before,
  .md-typeset .faq > summary::before {
    background-color: #FF8D1A;
    -webkit-mask-image: var(--md-admonition-icon--faq);
            mask-image: var(--md-admonition-icon--faq);
  }
```

![iShot_2024-04-26_19 40 05](https://github.com/squidfunk/mkdocs-material/assets/157043615/7078dc68-743c-4a72-801e-006fba3ad518)

**Comment by nimafanniasl** [comment]
Hey!
There is a problem with "details summary" ! Markdown doesn't get shown in it!

![image](https://github.com/squidfunk/mkdocs-material/assets/76901932/452937d8-6be0-464c-a544-00c345097c5b)
![image](https://github.com/squidfunk/mkdocs-material/assets/76901932/174007b7-5be0-4d0c-a161-31977fb4b3b1)

**Reply by alexvoss** [reply]
That is normal behavior. You can include Markdown in HTML if you use the [Markdown in HTML](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/?h=md+in+html#markdown-in-html) extension. Note that you need to add the `markdown` attribute to every HTML tag that surrounds the Markdown.

An alternative is to use the [new and experimental block syntax](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/html/) of the PymDown Extensions instead of writing raw HTML.

**Reply by nimafanniasl** [reply]
Thanks a lot!
I'm trying both of them out right now :)
